frontend/src/components/MapView/Layers/PointDataLayer/index.tsx

```diff
@@ -25,14 +28,50 @@ import {
 import { setEWSParams, clearDataset } from 'context/datasetStateSlice';
 import { createEWSDatasetParams } from 'utils/ews-utils';
 import { addPopupParams } from 'components/MapView/Layers/layer-utils';
+import {
+  CircleLayerSpecification,
+  FillLayerSpecification,
+  MapLayerMouseEvent,
+} from 'maplibre-gl';
+import { getLayerMapId, useMapCallback } from 'utils/map-utils';
+
+const onClick = ({
+  layer,
+  dispatch,
+  t,
+}: MapEventWrapFunctionProps<PointDataLayerProps>) => (
+  evt: MapLayerMouseEvent,
+) => {
+  addPopupParams(layer, dispatch, evt, t, false);
+
+  // TODO: maplibre: fix feature
+  const feature = evt.features?.find(
+    (x: any) => x.layer.id === getLayerMapId(layer.id),
```

Comment:
Same comments as above ( :footprints: and extracting the invariant function call from the `find` "loop")